### PR TITLE
fix(dynamic-import-vars): keep dynamic import assert arg intact

### PR DESCRIPTION
**Pull request by jorenbroekema**
## Rollup Plugin Name: `dynamic-import-vars`

This PR contains:

- [x] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

If yes, then include "BREAKING CHANGES:" in the first commit message body, followed by a description of what is breaking. 

List any relevant issue numbers: https://github.com/rollup/plugins/issues/1503

### Description

When transforming dynamic imports with variables in them, the second argument (import assertions) is stripped away and this information is lost. This Pull Request addresses that problem (see issue) by grabbing the argument Node and stringifying it and injecting it into the Magic String, so that this argument is kept as is.

This is needed for other plugins to correctly deal with imports that have import assertions, most notably https://github.com/jleeson/rollup-plugin-import-css which for CSS imports returns either a CSS string or CSSStyleSheet depending on whether or not an import assertion for CSS is present. If such an assertion is present, but dynamic-import-vars strips it away, the CSS plugin will bundle the CSS import as a CSS string rather than CSSStyleSheet, which would be different from local development, before bundling, since browsers natively do support import assertions now and return a CSSStyleSheet.

I might need some help with the testing, as it currently doesn't verify that the assertion is kept intact, I'm not sure how to test that, it seems in the final output[0].code the assertions are gone again, but when I test my changes in my local project, it does fix my issue.. and when I console.log the transform hook in this plugin with my changes, it does seem like it returns the correct code.

### CODE REVIEW

File: packages/dynamic-import-vars/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,354 @@
+{
```
Comment (shellscape) [inline packages/dynamic-import-vars/package-lock.json:1]: this file shouldn't exist

**Review by lukastaegert — COMMENTED** [review]
Looks mostly fine to me, but the comment about adding the acorn plugin makes me curious as this plugin is already added by Rollup.

**Comment by shellscape** [comment]
@jorenbroekema please give the replies here a look 🙏  if we reach 60 days without a reply, we'll consider the PR to be abandoned.

File: packages/dynamic-import-vars/README.md [hunk 2]
```diff
@@ -145,6 +145,29 @@ function importLocale(locale) {
...
+
+Will ensure the following import assertions remain intact:
+
+```js
+// Refer to rollup-plugin-import-css https://github.com/jleeson/rollup-plugin-import-css
+function importLocale(sheet) {
+  return import(`./styles/${sheet}.css`, { assert: { type: 'css' } });
+}
+```
+
```
[the remark this answers is not in the document]
Comment (jorenbroekema) [reply packages/dynamic-import-vars/README.md:170]: You're right, it works without needing to add it manually. I guess I based that off of some other plugin that was based on an older version of Rollup where this was still required.

File: packages/dynamic-import-vars/package-lock.json (hunk 1 above)
Comment (jorenbroekema) [reply packages/dynamic-import-vars/package-lock.json]: removed, I probably accidentally installed with npm instead of pnpm or something

**Comment by jorenbroekema** [comment]
Thanks for the reminder @shellscape , I addressed your comments, I might still need a hand on the test, the snapshot output seems to show the assert arg not appearing in the output, I'm not sure that's intended and I'm missing something, or my test is incorrect?

**Review by shellscape — APPROVED** [review]
I think that test is sufficient for now. Thanks for adding it. @lukastaegert please have another look.

**Review by lukastaegert — APPROVED** [review]
Looks good to me 👍

**Comment by shellscape** [comment]
Sorry about the delay getting this merged. Thanks again!

**Pull request merged**